pallets/compliance-manager/src/lib.rs

```diff
@@ -450,92 +451,85 @@ decl_event!(
 );
 
 impl<T: Config> Module<T> {
-    /// Fetches all claims of `target` identity with type
-    /// and scope from `claim` and generated by any of `issuers`.
-    fn fetch_claims<'a>(
-        target: IdentityId,
-        claim: &'a Claim,
-        issuers: &'a [TrustedIssuer],
-    ) -> impl 'a + Iterator<Item = Claim> {
+    /// Returns all non expired claims of type `ClaimType` and scope of `Scope` that were issued by
+    /// a trusted issuer for the given `id`.
+    fn fetch_claims(
+        id: IdentityId,
+        claim: &Claim,
+        trusted_issuers: &[TrustedIssuer],
+    ) -> Vec<Claim> {
```

Comment:
I think this should only need to return `Option<Claim>`.  The conditions only care if there is a claim from one of the trusted issuers or not.